### ISSUE TITLE
On single page site Sidebar does not highlight item navigated to (Docsify v5 Preview)

### ISSUE THREAD

**Issue by paulhibbitts** [docsifyjs/docsify#2509]
#### Steps to reproduce

View single page with Sidebar
Scroll down page, and tap on Header element
Copy resulting URL with id?
Open new Browser tab, paste copied URL

#### Current behaviour

The first item in the Sidebar remains highlighted

#### Expected behaviour

The item in the sidebar should be highlighted matching clicked id? on page (as is shown with Docsify v4, https://paulhibbitts.github.io/docsify-v4-single-page)

#### Other relevant information

- Docsify version:
- 
Most recent (v5) Preview

- [ ] Bug still occurs when all/other plugins are disabled?

- Docsify plugins (if the bug happens when plugins enabled, please try to isolate the issue):

#### Please create a reproducible sandbox

Here is a demo using the most recent Preview:

https://paulhibbitts.github.io/docsify-v5-preview-single-page/#

To view bug, go to site, scroll down page and click on a Header element, copy resulting URL, open new Browser tab and paste the URL Might be related to #2504 

[![Edit 307qqv236](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/307qqv236)

**Comment by Koooooo-7** [comment]
append:
Use the uri `https://paulhibbitts.github.io/docsify-v4-single-page/#/?id=strikethrough` directly to visit page.

Sometimes, it locates correctly but sometime, it locates on the `Bold` and highlight.

**Comment by paulhibbitts** [comment]
I've now created a sandbox to help with testing, which uses the latest Preview build: https://codesandbox.io/p/sandbox/long-waterfall-txknd8

**Comment by Koooooo-7** [comment]
> I've now created a sandbox to help with testing, which uses the latest Preview build: https://codesandbox.io/p/sandbox/long-waterfall-txknd8

Strange, I can not reproduce it now, although I revisit the https://paulhibbitts.github.io/docsify-v4-single-page/#/?id=strikethrough ...

**Comment by sy-records** [comment]
Reproduced in the sandbox.

<img width="871" alt="image" src="https://github.com/user-attachments/assets/0b807b10-2a74-4cfe-923b-951092f7f24a">

**Comment by Koooooo-7** [comment]
-- @sy-records  could you reproduce it steady?
I can not always reproduce it yesterday, sometimes fine, sometimes broken.
And today, I can not reproduce it totally... either in the sandbox or the site, or copying the sandbox file to locally test.
Sync what I checked, At the `#initHeadings`, it finds the `activeElement` as the , and marked it as `active`, and keep it event it gonna scrolling down to target id place.
https://github.com/docsifyjs/docsify/blob/ceb466ca9c29bec775f4ebda449f8ea40a5453df/src/core/event/index.js#L92

**Comment by sy-records** [comment]
> could you reproduce it steady?

yep, You can visit https://txknd8.csb.app/#/?id=bold which should have `bold` actived, but now has `My Open Publishing Space` actived.

<img width="659" alt="image" src="https://github.com/user-attachments/assets/dc9af5d7-d9c6-4f72-b234-382f04b058cd">
